clients/drcachesim/drpt2trace/pt2ir.cpp

```diff
@@ -270,150 +414,188 @@ pt2ir_t::convert(IN const uint8_t *pt_data, IN size_t pt_data_size, INOUT drir_t
         return PT2IR_CONV_ERROR_RAW_TRACE_TOO_LARGE;
     }
 
-    /* The Libipt decoder requires a fixed-size data block to be set for decoding before
-     * creation and does not offer an API to extend or modify the data being decoded. To
-     * decode multiple data chunks using a shared decoder, a workaround is needed. First,
-     * initialize an empty data block as a buffer. Next, each time we want to decode a
-     * chunk, reset the buffer and copy the data into it, and then set the decoder's
-     * decode position to the chunk's initial position.
-     */
+    /* Append the new data to the buffer. */
+    memcpy(pt_raw_buffer_.get() + pt_raw_buffer_data_size_, pt_data, pt_data_size);
+    pt_raw_buffer_data_size_ += pt_data_size;
+
+    if (!pt_decoder_has_sync_) {
```

Comment:
Add to comment the cases when pt_decoder_has_sync will be false.

Seems to me that it may be false only in the very beginning.